nixos/modules/services/web-servers/wildfly/default.nix

```diff
@@ -0,0 +1,169 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+
+  cfg = config.services.wildfly;
+
+  wildflyService = pkgs.stdenv.mkDerivation {
+    name = "wildfly-server";
+    builder = ./builder.sh;
+    inherit (pkgs) wildfly su;
+    inherit (cfg)
+      enable
+      cleanBoot
+      serverDir
+      user
+      group
+      userAdmin
+      passwordAdmin
+      initialHeapMem
+      maxHeapMem
+      initialMetaspaceSize
+      maxMetaspaceSize
+      wildflyConfig
+      wildflyMode
+      wildflyBind
+      wildflyBindManagement
+      ;
+  };
+
+  cleanupScript = pkgs.writeScript "wildfly-cleanup" ''
+    #!${pkgs.bash}/bin/bash
+    if [ "${toString cfg.cleanBoot}" == "1" ]; then
+      rm -rf ${cfg.serverDir}
+    fi
+  '';
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.wildfly = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable Wildfly.";
+      };
+
+      cleanBoot = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Delete and install the server instance file location every time. No persistence.";
+      };
+
+      serverDir = mkOption {
+        description = "Location of the server instance files";
+        default = "/opt/wildfly";
+        type = types.str;
+      };
+
+      user = mkOption {
+        default = "nobody";
```

Comment:
Use a dedicated user and group or use the [DynamicUser](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#DynamicUser=) systemd option